### ISSUE TITLE
CosineSimilarity outputs result smaller than -1

### ISSUE THREAD

**Issue by maybeLee** [keras-team/tf-keras#400]
**Describe the problem**.
As discussed in https://github.com/tensorflow/tensorflow/issues/58031, when I calculate the cosine similarity loss using `tf.keras.losses.CosineSimilarity` between two arrays: `[[0,2,3]]` and `[[0,2,3]]`, it outputs `-1.0000001`. Following the documentation: https://www.tensorflow.org/api_docs/python/tf/keras/losses/CosineSimilarity, the output should be within the range: [-1, 1].

**Describe the current behavior**.
The cosine similarity loss between `[[0,2,3]]` and `[[0,2,3]]` is out of [-1, 1].

**Describe the expected behavior**.
The cosine similarity loss between `[[0,2,3]]` and `[[0,2,3]]` is limited between [-1, 1].

**[Contributing](https://github.com/keras-team/keras/blob/master/CONTRIBUTING.md)**.

- Do you want to contribute a PR? (yes/no): yes
- If yes, please read [this page](https://github.com/keras-team/keras/blob/master/CONTRIBUTING.md) for instructions
- Briefly describe your candidate solution(if contributing):

**Standalone code to reproduce the issue**.
```
import tensorflow as tf
y_true = [[0.,2.,3.]]
y_pred = [[0.,2.,3.]]
cosine_loss = tf.keras.losses.CosineSimilarity()
res = cosine_loss(y_true, y_pred).numpy()
print("tensorflow: ", res)
```

**Source code / logs**.
```
tensorflow:  -1.0000001
```
